web/tests/test_dataset.py

```diff
@@ -9,3 +12,104 @@ def test_get_dataset_add(client_user_normal):
+@pytest.mark.parametrize('access_skip', [True, False])
+def test_dataset_wizard_form(client_user_normal,
+                             data_declaration_skip,
+                             storage_location_skip,
+                             legal_basis_skip,
+                             access_skip):
+    vip_user = UserFactory.create(groups=[VIPGroup()], first_name='Rebecca', last_name='Kafe')
+    storage_backend = StorageResourceFactory.create(name='test_backend', managed_by='test')
+    legal_basis_type = LegalBasisType(name='test_legal_basis', code='test')
+    legal_basis_type.save()
+
+    wizard_test_data = {
+        'dataset': [
+            {
+                'dataset-local_custodians': [vip_user.id],
+                'dataset-title': ['Hello Dataset'],
+                'dataset-project': [],
+                'dataset-comments': ['A comment'],
+                'dataset_wizard_view-current_step': ['dataset']
+            },
+            Dataset,
+        ],
+        'data_declaration': [
+            {
+                'data_declaration-title': ['Data declaration title'],
+                'data_declaration-type': ['2'],
+                'comment': ['Other provenance'],
+                'dataset_wizard_view-current_step': ['data_declaration'],
+                'data_declaration-skip_wizard': [data_declaration_skip],
+            },
+            DataDeclaration,
+        ],
+        'storage_location': [
+            {
+                'storage_location-category': ['master'],
+                'storage_location-backend': [storage_backend.id],
+                'storage_location-data_declarations': [],
+                'storage_location-datatypes': [],
+                'storage_location-location_description': ['hello'],
+                'dataset_wizard_view-current_step': ['storage_location'],
+                'storage_location-skip_wizard': [storage_location_skip],
+            },
+            DataLocation,
+        ],
+        'legal_basis': [
+            {
+                'legal_basis-data_declarations': [],
+                'legal_basis-legal_basis_types': [legal_basis_type.id],
+                'legal_basis-personal_data_types': [],
+                'legal_basis-remarks': ['Legal basis comment'],
+                'dataset_wizard_view-current_step': ['legal_basis'],
+                'legal_basis-skip_wizard': [legal_basis_skip]
+            },
+            LegalBasis,
+        ],
+        'access': [
+            {
+                'access-contact': [],
+                'access-user': [vip_user.id],
+                'access-project': [''],
+                'access-granted_on': [''],
+                'access-grant_expires_on': [''],
+                'access-access_notes': ['ssq'],
+                'dataset_wizard_view-current_step': ['access'],
+                'access-skip_wizard': [access_skip]
+            },
+            Access,
+        ],
+    }
+    for step_name, step_data in wizard_test_data.items():
```

Comment:
If you do not use `step_name`, you can call `wizard_test_data.values()` instead of `.items()`.

Also, can't the assertion be done in the other `for` loop before calling `post`. This would avoid looping twice